src/c/realize_c_type.c

```diff
@@ -787,8 +787,8 @@ static int do_realize_lazy_struct(CTypeDescrObject *ct)
         const struct _cffi_field_s *fld;
         PyObject *fields, *args, *res;
 
-        assert(!((ct->ct_flags & CT_IS_OPAQUE) ||
-                 (ct->ct_flags_mut & CT_UNDER_CONSTRUCTION)));
+        // should have been unset during init for the ctype
```

Comment:
I'm not sure I understand this comment. I'd think that the assert below is because opaque structs never have lazy field lists.